### PR TITLE
chore: Add docker workflow to github actions

### PR DESCRIPTION
**Pull request by onuzbee**
Closes : [DII-80](https://issues.corp.twilio.com/browse/DII-80)

# Fixes #

A short description of what this PR does.

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-cli/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please file a [support ticket](https://twilio.com/help/contact), or create a GitHub Issue in this repository.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -67,3 +67,40 @@ jobs:
...
+      - name: Fetch tag
+        id: get-tag
+        run: |
+          git fetch --tags
+          echo "::set-output name=TAG_NAME::$(git describe --tags $(git rev-list --tags --max-count=1))"
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
```
Comment (onuzbee) [inline .github/workflows/release.yml:86]: [Self] Added to github secrets

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -67,3 +67,40 @@ jobs:
...
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          context: ./
+          file: ./Dockerfile
+          push: true
+          tags: ${{ steps.meta.outputs.tags }}
+      - name: Image digest
+        run: |
+          echo ${{ steps.docker_build.outputs.digest }}
```
Comment (onuzbee) [inline .github/workflows/release.yml:106]: [Self] Prints sha digest for sanity

**Comment by onuzbee** [comment]
Sample successful workflow run on forked repo: https://github.com/onuzbee/twilio-cli/actions/runs/1212884537

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -67,3 +67,40 @@ jobs:
...
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v3
+        with:
+          images: twilio/twilio-cli
+          tags: type=semver,pattern={{version}}, value= ${{ steps.get-tag.outputs.TAG_NAME }}
```
[the remark this answers is not in the document]
Comment (onuzbee) [reply .github/workflows/release.yml]: Great suggestion! Updated

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -67,3 +67,40 @@ jobs:
...
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Docker meta
+        id: meta
```
[the remark this answers is not in the document]
Comment (onuzbee) [reply .github/workflows/release.yml]: Removed

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -67,3 +67,40 @@ jobs:
...
           is_append_body: true
           body: ${{ github.event.inputs.change-log }}
+
+  deploy-docker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check Out Repo
+        uses: actions/checkout@v2
+      - name: Fetch tag
```
[the remark this answers is not in the document]
Comment (onuzbee) [reply .github/workflows/release.yml]: Removed step

**Review by shrutiburman — APPROVED** [review]
Nicely done! 
LGTM! 🚀 👏

**Review by ravali-rimmalapudi — APPROVED** [review]
LGTM

**Pull request merged**